Replicator/ReplicatorOptions.hh

```diff
@@ -158,6 +176,127 @@ namespace litecore { namespace repl {
+            std::vector<uint8_t> path;
+            bool addScope = true;
+            if (FLSlice_Compare(spec.scope, kC4DefaultScopeID) == 0 && omitDefaultScope) {
+                addScope = false;
+            }
+            if (addScope) {
+                std::copy((uint8_t*)spec.scope.buf, (uint8_t*)spec.scope.buf + spec.scope.size,
+                          std::inserter(path, path.end()));
+                path.push_back('/');
+            }
+            std::copy((uint8_t*)spec.name.buf, (uint8_t*)spec.name.buf + spec.name.size,
+                      std::inserter(path, path.end()));
+            return {&path[0], path.size()};
+        }
+
+        static C4CollectionSpec collectionPathToSpec(alloc_slice path) {
+            const uint8_t* slash = path.findByte((uint8_t)'/');
+            slice scope = kC4DefaultScopeID;
+            slice name;
+            if (slash != nullptr) {
+                slice temp = slice {path.buf, static_cast<size_t>(slash - static_cast<const uint8_t*>(path.buf))};
+                scope = temp; //slice {path.buf, static_cast<size_t>(slash - static_cast<const uint8_t*>(path.buf))};
+                name = slice {slash + 1, path.size - scope.size - 1};
+            } else {
+                name = path;
+            }
+            return {name, scope};
+        }
+
+        inline static alloc_slice const kDefaultCollectionPath = collectionSpecToPath(kC4DefaultCollectionSpec, false);
+
+        struct CollectionOptions
+        {
+            alloc_slice                         collectionPath;
+
+            C4ReplicatorMode                    push;
+            C4ReplicatorMode                    pull;
+
+            fleece::AllocedDict                 properties;
+
+            C4ReplicatorValidationFunction      pushFilter;
+            C4ReplicatorValidationFunction      pullFilter;
+            void*                               callbackContext;
+
+            CollectionOptions(alloc_slice collectionPath_)
+            {
+                collectionPath = collectionPath_;
+            }
+
+            CollectionOptions(alloc_slice collectionPath_, C4Slice properties_)
+            : properties(properties_)
+            {
+                collectionPath = collectionPath_;
+            }
+
+            template <class T>
+            CollectionOptions& setProperty(fleece::slice name, T value) {
+                properties = Options::updateProperties(properties, name, value);
+                return *this;
+            }
+        };
+
+        std::vector<CollectionOptions> collectionOpts;
+
+        size_t collectionCount() const {
+            return collectionOpts.size();
+        }
+
+//        void deprecate() {
+//            *const_cast<Mode*>(&push) = kC4Disabled;
+//            *const_cast<Mode*>(&pull) = kC4Disabled;
+//            pushFilter = nullptr;
+//            pullValidator = nullptr;
+//            callbackContext = nullptr;
+//        }
```

Comment:
Don't include commented code like this

Reply:
I would use it to partially test the code flow, and I will need it before I completely plumb up everything.
I can save it and delete it from the PR.